File: pythonprogs/Stack.py
top = None
def isEmpty(stk):
    if stk == []:
        return True
    else:
        return False

def Push(stk, item):
    stk.append(item)
    top = len(stk) - 1

def Pop(stk):
    if isEmpty(stk):
        return "Underflow"
    else: 
        item = stk.pop()
        if len(stk) == 0:
            top = None
        else:
            top = len(stk) - 1
def Peek(stk):
    if isEmpty(stk):
        return "Undeflow"
    else:
        top = len(stk) - 1
        return stk[top]

def Display(stk):
    if isEmpty(stk):
        print("Stack is empty")
    else:
        top = len(stk) - 1
        print(stk[top], "<--top")
        for a in range(top-1, -1, -1):
            print(stk[a])
    

File: javaprogs/stack.java
class stack{
    private int stk[];
    private int top = -1;

    public stack(int length){
        stk = new int[length];
    }

    public void pop(){
        if(top == -1){
            System.out.println("Underflow Error! \t Cannot remove items from empty stack");
            return;
        }else{
            stk[top--] = 0;
        }
    }

    public void push(int item){
        if(top == stk.length - 1){    
            System.out.println("Overflow Error! \t");

        }else{
            stk[++top] = item;
        }
    }

    public int peek(){
        return stk[top];
    }

    public static void main(String args[]){
        stack obj = new stack(5);
        obj.push(10);
        obj.push(11);
        obj.push(12);
        obj.push(13);
        obj.push(14);
        obj.push(15);
        System.out.println(obj.peek());
        obj.pop();
        obj.pop();
        obj.pop();
        obj.pop();
        obj.pop();
        obj.pop();
    }
}